rust_src/src/marker.rs

```diff
@@ -39,6 +39,19 @@ impl LispMarkerRef {
     }
 }
 
+fn build_marker(buf: LispObject, charpos: ptrdiff_t, byte_pos: ptrdiff_t) -> LispObject {
+    let m = Lisp_Marker{
```

Comment:
I think we'd benefit from using the same assertions that are in the C. `eassert` in C is equivlaent to `debug_assert` in Rust.